### DISCUSSION TITLE
How to watch a control circuit, but not control it?

### DISCUSSION THREAD

**Discussion by pgorod** [arendst/Tasmota#12016] [Q&A] [answered]
Hi. I am exploring Tasmota with a few Sonoffs (POW R2, TH), using them to gather historical data from a water-heating system (solar + cylinder + gas burner).

So far, I have been able to avoid flashing (bought them flashed), compiling, and MQTT. I simply use a Scheduled task on a Windows PC on the LAN to gather the information with HTTP requests to the devices, and the rest is Excel.

I am quite happy with what I have achieved with the TH to get temperature readings every 5 minutes, but now I am facing another challenge and I would like to ask how you (the pros 😃 ) would recommend approaching.

There is a controller giving commands to the burner, to start it when certain conditions occur. I would like to have an historic record of this control signal (a 220V line comes on). Something like "yes, it was on for some period during the last 5m". But I shouldn't alter the existing function of the signal. 

Can I use the Sonoff POW R2 with Tasmota for this? How would I connect it, and how can I poll Tasmota to get the data?

This is probably simple, I'm just having trouble getting my head around it - the POW is meant to control the device whose current it measures, and it needs to be constantly powered itself. Can I wire it differently somehow to do what I need?

Thanks for your help!

P.S - if anybody's interested, I can share the Powershell script I use to poll Tasmotas from Windows. It's pretty basic but does the job.

**Comment by barbudor** [comment]
You can use the Sonoff POW with it's relay always switched ON (see `PowerOnState 4`)
Then you will use it only as a power-consumption measurement device
With some basic rule you can even "convert" the energy measurement into a more simple on/off state.

The thing that you do not explain is how you want to store the history ?
How and where do you want to present the information ?
Do you have a home automation system in which you can display that as graph or any other way ?

**Comment by ascillato** [comment, marked as answer]
>Can I use the Sonoff POW R2 with Tasmota for this? How would I connect it, and how can I poll Tasmota to get the data?

If you want to monitor the power consumption yes, you can use a sonoff POW in series with your burner. If you want just to monitor ON/OFF in a 220v line, you should use a Shelly 1 and connect in parallel the burner with the SW of the Shelly 1 and configure your Shelly to send its INPUT state only and not toggling its relay ( see rules in the docs )

If you connect your POW in series with your burner, you should configure your POW to not turn its relay to OFF and keep it always ON ( use command `poweronstate 4` in the Tasmota Console).

The data can be polled by using STATUS commands by MQTT or HTTP. See the docs for that.

**Comment by sfromis** [comment]
You are *severely* limiting you options by the crude technique of polling every 5 minutes. If using MQTT it would be piece of cake to get quick reporting of energy usage level changes. You can still use PowerDelta to get Tasmota to report when things happen, and then construct rules to pick up these data points. It would get a bit complex collecting data till next poll interval. You could also opt for a much shorter poll interval for at least better resolution.

**Comment by pgorod** [comment]
Thanks to all 3 of you for your answers!

Please don't worry about how I gather or process the information, I am no good with hardware and electricity, but I am a software developer and I had no problem working out that part. Of course one day I can go and learn MQTT and some home app, it's just that for now I am already dealing with several learning curves, I prefer to avoid those. My script gets the data from several devices and puts it into a single CSV, and I can see all that I need in Excel. 

About the wiring: I note the suggestion **using a Shelby** - thank you, if I can't get it working with the Sonoff that I already bought, I will certainly investage that device.

**About the wiring with the Sonoff**: so I have a live wire which is not always live (only when the control signal is there). So that doesn't seem to be a good place to connect the Sonoff inputs, otherwise it would be off until the control signal arrives, at which time it would have to power on, boot, connect to wi-fi before being operational. Is this what you were suggesting, do you think that could work well?

If not, I am guessing it would be better to just provide an independent power feed to the Sonoff inputs to keep it always on. But then how would I connect the control signal? It doesn't make sense to connect that to the Sonoff outputs.

**Reply by TrDA-hab** [reply]
I am just using PZEM-004t.
https://github.com/arendst/Tasmota/discussions/10567
https://github.com/arendst/Tasmota/discussions/11530

**Reply by barbudor** [reply]
You need to place the Sonoff before the relay that actually switch the burner on/off other wise the Sonoff will be powered along the burner.
Which is also an idea (detect when sonoff replies to polling requests...)

Other wise Adrian's idea to use a Shelly is the simplest without tinkering hardware

**Comment by pgorod** [comment]
Ok, so I guess the unspoken answer is: it's not possible with just a Sonoff POW (unless I'm willing to let it power down and up all the time, which doesn't seem very elegant).

Ideas were provided regarding using Shelly (because it has an additional SW input to switch the relay), and PZEM-004t (although that looks too difficult for my skills). 

I guess I will buy a Shelly. I promise to come back here and report on my solution, if I ever get one. Thanks for all your help! 👍

**Comment by pgorod** [comment]
Ok guys I Bought the Shelly, installed Tasmota very easily, OTA, with this excellent system: https://github.com/yaourdt/mgos-to-tasmota

My template is configured as follows:
![image](https://user-images.githubusercontent.com/15945027/118673721-258e0d80-b7f1-11eb-94d2-6db4bab75f05.png)

I got everything working like I wanted to in my work bench (as a proof-of-concept). This is basically all I need:

```
switchmode 1
rule1
   ON switch1#state=0 DO websend [10.0.0.201] zero ENDON
   ON switch1#state=1 DO websend [10.0.0.201] one ENDON
rule1 1
```
This effectively de-couples the SW input on the Shelly from the Relay. And it calls my web hooks whenever SW is turned on or off.

**I have only one problem:** when I went back to my hot-water burner I found that the line that I intended to connect to the SW input on the Shelly is not changing between 0 and 230V, as I thought... instead, it is around 108V when the burner is off, and 230V when it is on...

My question: can Tasmota deal with this? Can I specify somewhere what is the threshold, in Volts, for SW to be considered active?

If not, any of you electricity experts can point me to which extra hardware device I need to get something that I can feed to the SW input, in a way that Tasmota can handle it?

Thanks!

**Reply by sfromis** [reply]
The question here is not what Tasmota can deal with, but what the hardware can deal with. When it comes to switch inputs, Tasmota simply expects a high or low, as that is what a gpio pin supports (unless analog). And the hardware (ESP8266 chip) does not have thresholds. You might be able to modify or build a simple circuit to transform the input, but the details of that is outside of what I know much about.

**Reply by barbudor** [reply]
For a better support we would need to fully understand your circuit.
A good diagram that shows exact path of Line and Neutral wires and where exactly are located the burner and the relay that controls it is key to understand. Don't guess, look at the actual connections.
Include the shelly in your diagram and the exact points between which you are measuring 108V.

For now, my guess would be that the relay controlling the burner is on the neutral side and you measure a leak current from line through the burner. But I would prefer you to confirm that before advising a solution.

**Reply by pgorod** [reply]
I don't have a diagram because I don't have a circuit yet... I just tested the Shelly separately and now, when I was trying to figure out how to put in a circuit with the burner, I realized I have this difficulty. I will try to explain what I know about the burner, and how I measured. 

This will prove beyond any measure that I am a perfect fool with electricity... not my specialty. 

I have only one point on the burner where I can spot an electrical difference between burning/not burning. This was a tip given to me by the technician that came here some weeks ago to fix it.

![image](https://user-images.githubusercontent.com/15945027/118725400-cf888c80-b827-11eb-9f98-264551fb7625.png)

So, on that yellow plug, you see the red probe of my multi-meter sticking in. That's where the information I need is.

I set the multi-meter to "500V" and I put the black probe touching ground. That's what my right hand there is doing in this picture:

![image](https://user-images.githubusercontent.com/15945027/118725476-e9c26a80-b827-11eb-8dd4-ee6a3bd720e7.png)

You see the multi-meter measures 106V. When the burner starts burning, this changes to 230V which is what we use here in my country.

(BTW, if I swap red with black, it reads 215 when off... I don't know why...)

So my original plan was to connect a single wire to where the red probe is, into the Shelly SW input. To be honest, I haven't even tried it yet. It's not making sense... I don't understand where those 106V can be coming from, or why, and it seems the Shelly won't be able to use that anyway...

Do you have any advice on how I could put this together...? Thanks for your help and apologies for being such a noobie.

**Reply by barbudor** [reply]
If you can't draw a diagram of your existing installation, then you don't understand it.
If you don't understand it, don't tinker with it.
Mains is dangerous and can lead to electrocution or fire.

Your photos doesn't show clearly where is the Line, the Neutral, the relay and the burner.
You don't even show where the black probe is connected. Is that connected to the shelter ? so to the ground plane ?
You need to understand what each connection point are and follow the wires (they are colored!), draw and understand the whole

With neither you nor I knowing and understanding what your installation EXACTLY looks like, I won't give any advice that could be proven wrong and dangerous.
May be ask the help of the electrician.

**Reply by pgorod** [reply]
Yes the black probe is touching directly the "ground" wire coming from the wall socket. I just couldn't get it all in the same photo while holding my phone.

You are right that I don't understand this (yet!) and that I should be very careful before attempting anything. I don't think I will get many details about the internals of the controller and relays in the burner, I was hoping I could watch it without interfering in it. Maybe this isn't electrically possible.

If you look at my title: **How to watch a control circuit, but not control it?** I think it's actually a very interesting question in general. Or even **How to watch a control circuit, but not affect it in any way?**. I would find a use for these watchers in several of my devices (pumps, three-way valves, etc).

I might explore a very different possibility. There is a tiny window where we can peek into the burner and see the fire burning. Maybe a light sensor there could give me the information I need. And I wouldn't have to mess with the burner's circuit.

Thanks for joining this discussion, I consider that I am making progress, even if I can't get what I want, at least I am trying different things and scratching out the ones that can't work.

**Reply by barbudor** [reply]
> There is a tiny window where we can peek into the burner and see the fire burning

Oups I totally missed 3 key letters in your original post : "gas"
I was all about an electrical heater.
I don't own a gas burner so I have no idea how it works in details.
I imagine you need a bit of electricity to create the spark to light the gas but once the flamme is there, it should not use any electricity. So there is may be no electric level or consumption to monitor.

You may indeed better try to detect the light of the flamme or the temperature.
If you have a counter with digits on top, some people have used an ESP32 webcam + OCR to read the counter value (for water for example, would work with a gas counter too).

**Reply by ascillato** [reply]
Sometimes gas burners tests If there is flame just measuring if there is electrical continuity between the spark electrode and the pilot valve. A flame are just free electrons in the air, so they have a resistance close to zero. Flames are like electrical wires. If you measure continuity with a DMM over a flame, it will show that there is electrical connection.
That is the way that most gas burner works, but as we don't know how is your setup, we can't help you in a safe manner. So, please, call an electrician to do the automation you want. Mains and Gas, are not things to play with. Any modification you made to a burner (like as silly as putting a light sensor) will void your warranty and any insurance you could have over it.

**Reply by pgorod** [reply]
About the counter: I asked the gas company and they can actually give me the readings as pulses. BUT: in order to get a single wire inside the box where the counter is, to get their signal, I would have to use an entire certified security set-up with tons of requirements. So I can forget about it, much less try to get a camera with OCR in there...

About the burner... No warranty, no insurance. I also wouldn't say I am trying to make a modification. I am trying to make an observation of an existing system, without troubling it.

Q: How to watch a control circuit, but not affect it in any way?
A: Impossible, forget it.

😭

**Reply by sfromis** [reply]
To "watch" a control circuit (without some sort existing indicator to snoop on), you'll have to connect to it. To do that you need to *exactly* know and understand the electrical behavior, and if it is anywhere near mains voltage level, keep away without having that knowledge before or after examination of the device.

Sometimes, you may be able to monitor the power supply, as many devices will use power to control operation. Maybe not a lot, but patterns in small variation may be very informative.

**Reply by sfromis** [reply]
With a burner, there would often be some type of noise/vibrations to detect.

**Comment by lalo-uy** [comment]
That gas burner do not have a blower? Yo my monitor the blower current.

**Comment by pgorod** [comment]
Just to leave some feedback here, I ended up with a pretty nice solution to watch (literally, watch as in _see_) my gas burner burning, using a small cheap circuit called XH-M131, with a long extension added to the wire connecting the "eye" (light sensor). This way everything stays outside the burner area except the sensor, which is in a plastic tube right in front of the little glass window where we can see the fire burning. After some tweaking of the sensitivity it gets perfect results switching on when the gas burns, and off when it doesn't.

This signal feeds into a Shelly1PM (it didn't have to be a PM, but that's what I had), into the SW connection. The Shelly then notifies a local web server with `websend`. 

I also use the same Shelly, with the relay working independently of all this, to turn on/off the power to the burner, to establish a smart time schedule that can take into account many factors, since I am controlling it from my own PHP program in a PC. For example, I plan to use current local weather data, obtained from the web, to make decisions based on whether it's sunny or not, to take better advantage of the solar panels connected to that system.

I'm doing this mostly for fun, but there are significant financial gains to be made from an improved energy management... oh and I'm saving the planet, of course 😊
